### PR TITLE
Prepare Docker setup for monorepo

### LINKED ISSUE #865: Move Docker-only directories from root to `docker/`
## Problem

The repository contains a few directories that are only configuring third-party services that are orchestrated via Docker Compose.

- `nginx/`
- `postgres/`
- `plausible/` (proposed #804)
- `clickhouse/` (proposed #804)

## Description

Taking the approach from the `WordPress/openverse-catalog` repo, these can be moved into a root `docker/` folder for better management.

### LINKED ISSUE #866: Use profiles in Docker Compose
## Problem

The Docker Compose file starts a lot of services that may not always be necessary.

## Description

Using [Docker Compose profiles](https://docs.docker.com/compose/profiles/), we can enable a subset of services to be spun up, as needed. This becomes even more useful when we add even more services from the catalog such as the 2 Airflow instances.

## Alternatives

Splitting up the Docker Compose file into smaller files, or launching individual services by name are options but using profiles is the most ergonomic option.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #865 by @dhruvkb 

## Description

This PR updates the Docker setup for the API to make it modular, and more accommodating of the Docker setup of the catalog, when merged.

Changes:
- `postgres/` config moved to `docker/db`.
  - The env file pertaining to Plausible moved to `docker/plausible_db`.
- `clickhouse` config moved to `docker/clickhouse`.
- `nginx/` config moved to `docker/nginx`.
- Elasticseach env vars from Compose config and `just` recipes from root `justfile` extracted to `docker/es`.
- `upstream_db` config "lifted" from WordPress/openverse-catalog. This has several added benefits:
  - Represents actual nature of `upstream_db` instead of hacks.
  - Includes `pgcli` in the Docker image.
  - Updates PostgreSQL version from 13.2 to 13.6.
- `ingestion_server` cleanup removes now unnecessary mock schemas and mock data.
- `load_sample_data.sh` script simplified because upstream DB gets automatically migrated using included initialisation SQL files.
  - Also removed a redundant copying of providers from `db` to `upstream_db`
  - Use of Bash functions to reduce duplication in `load_sample_data.sh`.
  - Repeated `-U deploy -d openledger` flags made unnecessary by setting `PGDATABASE` and `PGUSER` environment variables.
- `subprocess.run` calls to Docker Compose extracted to a util to reduce duplication in ingestion server test.
- `sample_images.csv` renamed to `sample_image.csv` to use interpolation with table name.

## Testing Instructions

CI should continue to pass as it an internal refactor.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
@krysal fair point, I dropped all changes pertaining to #866 from this PR.

**Comment by dhruvkb** [comment]
Upping priority to high as it blocked #866.

**Review by AetherUnbound — CHANGES_REQUESTED** [review]
This is fantastic! The code changes look good, but I can't build due to the following error:

```
j build
just dc build 
docker-compose -f docker-compose.yml build
unexpected character "-" in variable name near "http.cors.allow-origin=\"/.*/\"\n"
error: Recipe `dc` failed on line 117 with exit code 15
error: Recipe `build` failed on line 121 with exit code 15
```

I suspect this has something to do with the `docker/es/env.docker` file?

**Comment by dhruvkb** [comment]
@AetherUnbound I cannot repro this locally. `just dc build` worked normally for me. What version of Compose are you using? Profiles are only supported by [Compose v2](https://docs.docker.com/compose/compose-v2/) (the Docker plugin) and not Compose v1 (the standalone tool).

**Comment by AetherUnbound** [comment]
This is now running just fine on my desktop 🤨 You've got to be right, there's some weird version mismatch or something. Here's my desktop (working) report:

```
$ docker-compose version
docker-compose version 1.29.2, build unknown
docker-py version: 5.0.0
CPython version: 3.8.13
OpenSSL version: OpenSSL 1.1.1q  5 Jul 2022
```

**Review by AetherUnbound — APPROVED** [review]
Confirmed that this works on my desktop! I'll post my laptop details once I'm on it again to see what might be the cause of the failure.

**Comment by AetherUnbound** [comment]
Hmm, it's using `docker-compose` locally for me 🤔 
```

$ docker-compose version
Docker Compose version 2.15.0

$ docker compose version
Docker Compose version v2.16.0

$ which docker-compose
/home/linuxbrew/.linuxbrew/bin/docker-compose
```

I just ran `brew upgrade docker-compose` and now it's working! How odd! Sorry for the confusion 🙂

File: load_sample_data.sh [hunk 1]
```diff
@@ -5,6 +5,28 @@ CACHE_SERVICE_NAME="${CACHE_SERVICE_NAME:-cache}"
...
+# Load sample data
+function load_sample_data {
+  docker-compose exec -T "$UPSTREAM_DB_SERVICE_NAME" /bin/bash -c "psql <<-EOF
+    DELETE FROM $1;
+		\copy $1 \
+			from './sample_data/sample_$1.csv' \
+			with (FORMAT csv, HEADER true);
+		REFRESH MATERIALIZED VIEW $1_view;
+		EOF"
+}
```
Comment (krysal) [inline load_sample_data.sh:21]: Nice, this file is taking quite a good form.

**Review by krysal — APPROVED** [review]
Very neat to have `justfile`s by services 💯

File: sample_data/sample_audio.csv
[the remark this answers is not in the document]
Comment (dhruvkb) [reply sample_data/sample_audio.csv:1]: Earlier we were populating `image_view` from sample data. Now we load `image` from sample data and refresh the matview. Same from audio.

**Pull request merged**